src/Arcus.Testing.Integration.DataFactory/TemporaryDataFlowDebugSession.cs

```diff
@@ -389,22 +404,57 @@ public class RunDataFlowOptions
 
         internal Collection<string> LinkedServiceNames { get; } = new();
         internal IDictionary<string, BinaryData> DataFlowParameters { get; } = new Dictionary<string, BinaryData>();
+        internal IDictionary<string, IDictionary<string, object>> DataSetParameters { get; } = new Dictionary<string, IDictionary<string, object>>();
 
         /// <summary>
         /// Adds a parameter to the DataFlow to run.
         /// </summary>
         /// <exception cref="ArgumentException">Thrown when the <paramref name="name"/> is blank.</exception>
+        /// <exception cref="ArgumentNullException">Thrown when the <paramref name="value"/> is null.</exception>
         public RunDataFlowOptions AddDataFlowParameter(string name, object value)
         {
             if (string.IsNullOrWhiteSpace(name))
             {
                 throw new ArgumentException("DataFlow parameter name should not be blank", nameof(name));
             }
 
+            ArgumentNullException.ThrowIfNull(value);
+
             DataFlowParameters[name] = BinaryData.FromObjectAsJson(value);
             return this;
         }
 
+        /// <summary>
+        /// Adds a parameter to a DataSet that is part of the targeted DataFlow.
+        /// 
+        /// "datasetName" should be the "Output stream name" of the source or sink dataset in the DataFlow, rather than the actual DataSet name.
+        /// </summary>
+        /// <exception cref="ArgumentException">Thrown when the <paramref name="datasetName"/> is blank.</exception>
+        /// <exception cref="ArgumentException">Thrown when the <paramref name="parameterName"/> is blank.</exception>
+        /// <exception cref="ArgumentNullException">Thrown when the <paramref name="parameterValue"/> is null.</exception>
+        public RunDataFlowOptions AddDataSetParameter(string datasetName, string parameterName, object parameterValue)
```

Comment:
@stijnmoreels `datasetName` param **name** does not really reflects what it is, since this parameter represents either the source or sink name. 
Shall we leave this name or do you have a suggestion for a better naming please? 
`sourceOrSinkName` maybe?

Reply:
Renamed param to `sourceOrSinkName`

Reply:
Applied change in [#09c8cb](https://github.com/arcus-azure/arcus.testing/pull/228/commits/09c8cb9620589a06ff65092cad26fa5840f24a10)